NuRadioReco/modules/RNO_G/triggerBoardResponse.py

```diff
@@ -0,0 +1,357 @@
+                msg += f" fields set. Ensure that the following settings are in your Detector configuration {self._mandatory_fields}"
+                self.logger.warn(msg)
+
+        self.logger.debug(f"Found trigger channels {found_trigger_channels}")
+        self.logger.debug(f"Found trigger amps {trigger_amp_response_name}")
+
+        def trigger_filter(freqs):
+            return trigger_amp_response["gain"](freqs) * trigger_amp_response["phase"](freqs)
+
+        return found_trigger_channels, trigger_filter
+
+    def digitize_trace(self, station, det, trigger_channels, vrms):
+        for channel_id in trigger_channels:
+            channel = station.get_channel(channel_id)
+
+            digitized_trace, adc_sampling_frequency = self._adc.get_digital_trace(
+                station,
+                det,
+                channel,
+                Vrms=vrms,
+                trigger_adc=True,
+                adc_type="perfect_floor_comparator",
+                trigger_filter=None,  # Applied already
+                clock_offset=self._clock_offset,
+                adc_output=self._adc_output,
+                return_sampling_frequency=True,
+                channel_id=channel_id - self.__channel_offset,
+            )
+
+            channel.set_trace(digitized_trace, adc_sampling_frequency)
+
+    @register_run()
+    def run(self, evt, station, det, requested_channels=[], vrms=None, apply_adc_gain=True,
+            digitize_trace=True, channel_offset=0, do_apply_trigger_filter=True):
+        """
+        Applies the additional filters on the trigger board and performs a gain amplification
+        to get the correct number of trigger bits.
+        This module expects that all of the trigger channels (either defined by the existance of
+        `trigger_XXX` fields in the detector .json file or via the `requested_channels`
+        argument) are identical.
+
+        Parameters
+        ----------
+        evt : Event
+            Event to run the module on
+        station : Station
+            Station to run the module on
+        det : Detector
+            The detector description
+        requested_channels : list (default: [])
+            Channels that this module should consider when applying the filter
+        vrms : float (default: None)
+            The Vrms of the trigger channels including the trigger board filters
+            If set to `None`, this will be estimated using the waveforms
+        apply_adc_gain : bool (default: True)
+            Apply the gain shift to achieve the specified level of noise bits
+        digitize_trace : bool (default: True)
+            Apply the quantization to the voltages (uses `NuRadioReco.modules.analogToDigitalConverter` to do so)
+        channel_offset : int (default: 0)
+            The offset to apply to the channel IDs when requesting the detector description. This is
+            necessary if you have created a copy of the original channel and changed the ID.
```

Comment:
btw @colemanalan, I also added this hack which I think we can remove. It was handy for a hack which we should not carry into the future. Sorry

Reply:
removed!